### PR TITLE
Bluetooth: SMP: Fix pairing using debug keys

### PR DESCRIPTION
**Pull request by joerchan**
Fix issue when checking if SMP pairing procedure is allowed to use the
SMP debug keys. This check did not consider the case where the keys
pointer was assigned, but did not contain a valid LTK key.
This resulted in being unable to pair with debug keys without an
existing bond.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/host/smp.c [hunk 1]
```diff
@@ -616,7 +616,7 @@ static u8_t get_encryption_key_size(struct bt_smp *smp)
 /* Check that if a new pairing procedure with an existing bond will not lower
  * the established security level of the bond.
  */
-bool update_keys_check(struct bt_smp *smp)
+static bool update_keys_check(struct bt_smp *smp)
```
Comment (jhedberg) [inline subsys/bluetooth/host/smp.c]: This change seems unrelated? It's a good one though - just deserves its own commit, I think.

Comment (joerchan) [reply subsys/bluetooth/host/smp.c]: Unrelated yes, Thought it small and harmless enough that it didn't need it's own.
Split it?

Comment (jhedberg) [reply subsys/bluetooth/host/smp.c]: Doesn't need its own PR, but I'd still try to keep the commits clean, so that the content matches the message, i.e. create a separate commit for this one.

Comment (joerchan) [reply subsys/bluetooth/host/smp.c]: Moved to it's own commit.

**Review by jhedberg — APPROVED** [review]
